wallet/src/main/java/com/android/identity_credential/wallet/ui/destination/provisioncredential/EvidenceRequest.kt

```diff
@@ -760,28 +761,29 @@ fun AusweisView(
             }
         }
         composable(AusweisModel.Route.PIN_ENTRY.route) {
-            val pin = remember { mutableStateOf("") }
             Column {
                 Row(
                     modifier = Modifier.fillMaxWidth(),
                     horizontalArrangement = Arrangement.Center
                 ) {
-                    TextField(value = pin.value, onValueChange = {pin.value = it},
-                        label = { Text(stringResource(R.string.eid_enter_pin)) })
+                    Text(
+                        text = stringResource(R.string.eid_enter_pin),
+                        textAlign = TextAlign.Center,
+                        modifier = Modifier.padding(8.dp),
+                        style = MaterialTheme.typography.titleLarge
+                    )
                 }
                 Row(
                     modifier = Modifier.fillMaxWidth(),
                     horizontalArrangement = Arrangement.Center
                 ) {
-                    Button(
-                        enabled = status.value is AusweisModel.WaitingForPin &&
-                                pin.value.matches(Regex("^\\d{6}\$")),  // 6-digit
-                        onClick = {
-                            model.providePin(pin.value)
-                            pin.value = ""
+                    PassphraseEntryField(
+                        constraints = PassphraseConstraints.PIN_SIX_DIGITS,
+                        checkWeakPassphrase = false
+                    ) { passphrase, meetsRequirements, donePressed ->
+                        if (meetsRequirements && donePressed) {
+                            model.providePin(passphrase)
```

Comment:
I think this should be just `if (meetsRequirements)` - that way the user will go to the next screen as soon as they have entered the sixth and final digit.

Reply:
Done